### ISSUE TITLE
Token cache in id_token mutator not invalidated on JWKS rotation

### ISSUE THREAD

**Issue by DimitarPetrov** [ory/oathkeeper#763]
**Describe the bug**

Currently, we use oathkeeper's id_token mutator to issue tokens for internal communication. Also, we have a job which rotates its JWKS for security reasons. The way this job works is as described here: https://github.com/ory/oathkeeper/issues/253

What we observed is that when the job for JWKS rotation finishes, all client requests fail with 401 for duration something around the TTL of the tokens. Once we increased the TTL the issue was much more observable.

After some investigation, it looks like the id_tokens are cached due to performance optimization, and this cache is not invalidated once the JWKS is rotated. Then when a client having a token in the cache performs a subsequent call he receives 401 since the token is not issued with the current JWKS that is served by oathkeeper (it is signed with the previous one).

**Reproducing the bug**

1. Configure oathkeeper rule with id_token mutator with a long enough TTL (ex. 1h)
2. Make a call matching this rule in order for oathkeeper to cache your id_token.
3. Rotate the JWKS of oathkeeper like proposed here: https://github.com/ory/oathkeeper/issues/253
4. Make the same call as in 2. and the token received won't have a valid signature if validated with the most recent JWKS served by oathkeeper.

**Expected behavior**

I would expect oathkeeper to detect the new JWKS and invalidate the token cache that contains tokens signed with the old JWKS.

**Comment by aeneasr** [comment]
Yup, the cache should definitely be invalidated in that scenario. PRs highly welcomed!
